### PR TITLE
Update performance benchmarks in README.md

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Update performance benchmarks in README.md

## Commits

- docs: update performance benchmarks in README.md

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -315,21 +315,15 @@ Job=ShortRun  IterationCount=3  LaunchCount=1
...
-| NewUlid      | Factory.NewUlid         | PseudoRandom       |  63.25 ns |  2.600 ns | 0.142 ns |  0.11 |    0.00 | 0.0024 |      40 B |          NA |
-| NewUlid      | Ulid.NewUlid            | PseudoRandom       |  63.75 ns |  2.418 ns | 0.133 ns |  0.11 |    0.00 | 0.0024 |      40 B |          NA |
-| NewUlid      | Guid.NewGuid            | PseudoRandom       | 592.70 ns | 12.714 ns | 0.697 ns |  1.00 |    0.00 |      - |         - |          NA |
+| Type         | Method                  | RandomProviderType | Mean     | Error    | StdDev   | Gen0   | Allocated |
+|------------- |------------------------ |------------------- |---------:|---------:|---------:|-------:|----------:|
+| ParseUlid    | Ulid.Parse(StringUtf16) | ?                  | 65.91 ns | 0.558 ns | 0.522 ns | 0.0024 |      40 B |
+| UlidToString | Ulid.ToString           | ?                  | 48.28 ns | 0.251 ns | 0.222 ns | 0.0048 |      80 B |
+| ParseUlid    | Ulid.Parse(StringUtf8)  | ?                  | 58.05 ns | 0.315 ns | 0.294 ns | 0.0024 |      40 B |
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply README.md:320]: OK
